e2e/cypress/integration/channel/convert_channel_to_private_spec.js

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @channel
+
+describe('Channels', () => {
+    let testUser;
+    let testTeam;
+
+    before(() => {
+        cy.apiInitSetup().then(({team, user}) => {
+            testUser = user;
+            testTeam = team;
+        });
+    });
+
+    it('MM-28274 - Convert to private channel should only be shown to users with permission', () => {
```

Comment:
@srkgupta I have added E2E tests here, it tests 4 different things
- Regular system user with the new permission _can_ convert public channels to private
- Regular system user _without_ the new permission _cannot_ convert public channels to private
- Team admin without the permission _cannot_ convert public channels to private
- Team admin with the permission _can_ convert public channels

Reply:
Awesome. Thanks @fmunshi